src/bazel/tasks.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as vscode from "vscode";
+import { IBazelCommandOptions } from "./bazel_command";
+
+/**
+ * Returns a {@code ShellQuotedString} indicating how to quote the given flag
+ * if it contains spaces or other characters that need escaping.
+ */
+function quotedOption(option: string): vscode.ShellQuotedString {
+  return { value: option, quoting: vscode.ShellQuoting.Strong };
+}
+
+/**
+ * Creates a new task that invokes a build or test action.
+ *
+ * @param command The Bazel command to execute.
+ * @param options Describes the options used to launch Bazel.
+ */
+export function createBazelTask(
+  command: "build" | "test",
+  options: IBazelCommandOptions,
+): vscode.Task {
+  const args = [command as string]
+    .concat(options.targets)
+    .concat(options.options)
+    .map(quotedOption);
+
+  let commandDescription: string;
+  switch (command) {
+    case "build":
+      commandDescription = "Build";
+      break;
+    case "test":
+      commandDescription = "Test";
+      break;
+  }
+
+  const targetsDescription = options.targets.join(", ");
+  return new vscode.Task(
+    { type: "bazel", command, targets: options.targets },
+    // TODO(allevato): Change Workspace to Global once the fix for
```

Comment:
I don't suppose the vscode version is expose in a way to test to the code can be ready in advance?

Reply:
I'd have to check to see if there are recommended best practices for how to toggle features based on version, or if there is a way to just require a particular minimum version of VS Code for the extension (it may be tied to the `vscode` version used in package.json).